### PR TITLE
Support additional platforms

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Build for `darwin-arm64`, `linux-arm64`, and `windows-amd64` (the latter mostly for completeness). None of these platforms are necessarily supported at this time.

See https://github.com/mattermost/mattermost-plugin-playbooks/pull/868 for more background.

#### Ticket Link
N/A

### CODE REVIEW

**Review by streamer45 — COMMENTED** [review]
Thanks @lieut-data for these. I've tested on my MacBook and I am still getting some integrity errors. I've both run a `make clean` and cleared npm cache but to no avail:

Screenshot 2021-12-10 at 11 48 33

Running `node 16.9.1` and `npm 7.21.1`.

On a separate note, I have one small (I hope) ask: the reason no other platforms were originally added was that re-building the plugin takes some considerable time which in turn makes deployment a longer process. I am wondering if we could find a simple way (e.g. through an env var) to let only a specified arch to get built if set. This would allow developers to only build for their own platform during code-debug cycles.

**Comment by lieut-data** [comment]
> Thanks @lieut-data for these. I've tested on my MacBook and I am still getting some integrity errors. I've both run a make clean and cleared npm cache but to no avail:

@streamer45, sorry about the churn here: I /thought/ I had tested this correctly in my environment, but I could easily reproduce what you observed. @cpoile ran into this for Playbooks too with his M1, so I missed that there as well.

I've changed the code to just strip out the invalid integrity checks in the `Makefile` instead. Can you try again?

**Comment by streamer45** [comment]
> I've changed the code to just strip out the invalid integrity checks in the `Makefile` instead. Can you try again?

Thanks Jesse, working as expected now :+1:

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+16.4.0
```
[the remark this answers is not in the document]
Comment (streamer45) [reply .nvmrc:1]: I am not sure cause I vaguely recall settling on that version for a reason. Given it's newer I'd be in favor of keeping it as such. If we get into issues we can always downgrade.

[the remark this answers is not in the document]
Comment (lieut-data) [reply .nvmrc]: Done!

**Comment by lieut-data** [comment]
Looking into the CircleCI stuff now.

**Comment by lieut-data** [comment]
For now, I just ripped out `.nvmrc`. I think we switched to using a custom executor for CircleCI on Playbooks since the custom one we compiled didn't have everything we needed.

**Comment by streamer45** [comment]
> For now, I just ripped out `.nvmrc`. I think we switched to using a custom executor for CircleCI on Playbooks since the custom one we compiled didn't have everything we needed.

That's totally fine. I had to setup it this way to get latest Golang (1.17) but also the expected node version. We can revisit in the future.

**Comment by lieut-data** [comment]
@streamer45, does it make sense to merge this one?

**Comment by streamer45** [comment]
> @streamer45, does it make sense to merge this one?

@lieut-data 

I implicitly assumed you were going to address [my concern](https://github.com/mattermost/mattermost-plugin-calls/pull/3#pullrequestreview-828676500) over build times. It can be a separate PR if you prefer.

On a separate note, I've started to wonder if it would be simpler to just remove the integrity checksum for the affected dependencies from `package-lock.json` rather than the rewriting script hack. I don't think we'll be updating those any time soon so it may be a cleaner approach. What do you think?

**Comment by lieut-data** [comment]
> I implicitly assumed you were going to address my concern over build times. It can be a separate PR if you prefer.

Oops, I never saw this comment! My bad :) @agarciamontoro actually implemented exactly your proposal in https://github.com/mattermost/mattermost-plugin-playbooks/pull/876. Thoughts on effectively cherry-picking that as a solution in a separate PR?

> On a separate note, I've started to wonder if it would be simpler to just remove the integrity checksum for the affected dependencies from package-lock.json rather than the rewriting script hack. I don't think we'll be updating those any time soon so it may be a cleaner approach. What do you think?

One consideration is that I believe each `npm install` would add those integrity checks back, so any new contributors might inadvertently re-submit the changes. (We ran into exactly this for Playbooks before we expanded the "hack" :/)

**Comment by streamer45** [comment]
> @agarciamontoro actually implemented exactly your proposal in [mattermost/mattermost-plugin-playbooks#876](https://github.com/mattermost/mattermost-plugin-playbooks/pull/876). Thoughts on effectively cherry-picking that as a solution in a separate PR?

That would be awesome :+1: 

> One consideration is that I believe each `npm install` would add those integrity checks back, so any new contributors might inadvertently re-submit the changes. (We ran into exactly this for Playbooks before we expanded the "hack" :/)

Interesting cause I tried rebuilding `node_modules` from scratch a couple of times and didn't get shasums replaced. Maybe bumping NPM/Node version would do it.

**Review by streamer45 — APPROVED** [review]
No need to block this further, we can merge as is and improve if needed, thanks again :+1:

**Pull request merged**